java-backend/src/main/java/org/kframework/backend/java/kil/Definition.java

```diff
@@ -197,6 +199,10 @@ public void addKoreRules(Module module, GlobalContext global) {
         List<org.kframework.definition.Rule> koreRules = JavaConversions.setAsJavaSet(module.rules()).stream()
                 .filter(r -> !r.att().contains(AUTOMATON))
                 .collect(Collectors.toList());
+
+        //Ensures that rule order in all collections remains the same across across different executions.
+        koreRules.sort(Comparator.comparingInt(rule -> rule.body().hashCode()));
```

Comment:
This is not going to guarantee a unique order as two rules can have the same body (and thus the same hash code) but different side conditions.

Reply:
That's right. But it still reduces non-determinism enough to eliminate the issues I mentioned above.